src/unit_tests/shared/test_agent_op.c

```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
     cJSON_Delete(payload);
 }
 
+static void test_get_agent_group(void **state) {
```

Comment:
We should divide this test in three different ones. i.e.:
**test_get_agent_group_success()**
,
**test_get_agent_group_missing_agent_info()**
and
**test_get_agent_group_missing_group()**
In this way, each different scenario will have its own test and will be easier to sustain.